astropy/io/fits/tests/test_header.py

```diff
@@ -617,12 +617,9 @@ def test_missing_keyword(self):
         """Test that accessing a non-existent keyword raises a KeyError."""
 
         header = fits.Header()
-        pytest.raises(KeyError, lambda k: header[k], 'NAXIS')
```

Comment:
Why remove this?

Reply:
This is testing if `header['NAXIS']` raises a `KeyError`, i.e. exactly what's done below anyway when checking the error message (correct me if I'm wrong).

Reply:
Depends a bit if this has side-effects. In my experience changing such things requires careful analysis of the test and the tested code and seldom worth the trouble. Since this is `io.fits` I'm doubly careful 😄 

I would prefer keeping the duplicated test just to make sure. @saimn what do you think?

Reply:
I think it's fine to remove the first check in this case, as I understand it the check was there only to make sure that the correct exception is raised.

Reply:
Nothing against keeping this in for erring on the safe side here. Though if we were serious about safeguarding against different behaviour of the wrapper function vs. context forms, we'd probably have to add the extra check to every `pytest.raises` test...